packages/php-wasm/logger/src/lib/logger.ts

```diff
@@ -12,309 +15,159 @@ export type LogSeverity = 'Debug' | 'Info' | 'Warn' | 'Error' | 'Fatal';
 /**
  * Log prefix.
  */
-export type LogPrefix = 'Playground' | 'PHP-WASM';
+export type LogPrefix = 'PHP-WASM' | 'PHP' | 'JavaScript';
 
 /**
  * A logger for Playground.
  */
 export class Logger extends EventTarget {
 	public readonly fatalErrorEvent = 'playground-fatal-error';
 
-	/**
-	 * Log messages
-	 */
-	private logs: string[] = [];
-
-	/**
-	 * Whether the window events are connected.
-	 */
-	private windowConnected = false;
-
-	/**
-	 * The length of the last PHP log.
-	 */
-	private lastPHPLogLength = 0;
-
 	/**
 	 * Context data
 	 */
 	private context: Record<string, any> = {};
 
-	/**
-	 * The path to the error log file.
-	 */
-	private errorLogPath = '/wordpress/wp-content/debug.log';
-
-	constructor(errorLogPath?: string) {
+	// constructor
+	constructor(
+		// Log handlers
+		private readonly handlers: Function[] = []
+	) {
 		super();
-		if (errorLogPath) {
-			this.errorLogPath = errorLogPath;
-		}
 	}
 
 	/**
-	 * Read the WordPress debug.log file and return its content.
-	 *
-	 * @param UniversalPHP playground instance
-	 * @returns string The content of the debug.log file
+	 * Get all logs.
+	 * @returns string[]
 	 */
-	private async getRequestPhpErrorLog(playground: UniversalPHP) {
-		if (!(await playground.fileExists(this.errorLogPath))) {
-			return '';
-		}
-		return await playground.readFileAsText(this.errorLogPath);
+	public getLogs(): string[] {
```

Comment:
Should this be now done in the `log-to-memory.ts` handler?

Reply:
I think that the logger should keep the method. As far as users are concerned, they don't need to care about handlers, they should only interact with the logger. 

I added an error message ([89b58bc](https://github.com/WordPress/playground/pull/1264/commits/89b58bc349005c3729366b229dc9e2d673e5d9bc)) in case someone has a custom instance without the logTo Memory handler.

Initially, I thought we should have different log consumers like memory and file, but the [log to file exploration](https://github.com/WordPress/playground/pull/1292) made me realize that we need memory logging as the fundamental storage, so we can assume that it will always be there.

If that changes over time, the user should still only interact with the logger, and consumers/handlers should manage data fetching internally.

Reply:
Let's document the rationale in the docstring for posterity.